src/main/java/com/amihaiemil/zold/RtWallet.java

```diff
@@ -113,7 +114,7 @@ public double balance() throws IOException{
     @Override
     public void pay(
         final String keygap, final String user,
-        final double amount, final String details
+        final String amount, final String details
```

Comment:
@ammaratef45 this change will affect JSON payload format, type `amount` changed from `JSON-number` to `JSON-string`:
before:
```json
{
  "amount": 1234.56
}
```
after:
```json
{
  "amount": "1234.56"
}
``` 

I'm not sure here (because I didn't find API docs for JSON payload), but most probably API won't accept such format.

Reply:
@g4s8 reverting it makes the issue meaningless, what do you think? we may need to rediscuss the issue, right?

Reply:
@ammaratef45 you can use `json.add("amount", new BigDecimal(amount))` - that code adds amount as json number, but with arbitrary precision

Reply:
@g4s8 The problem is that this change meant to be used for avoiding any casting or precision problems and let the higher layers handle this.

Reply:
@ammaratef45 but `BigDecimal` doesn't limit the precision, so I think it's safe to use here. In any case we should conform to wts API protocol, so if you have any doubts you can submit a ticket to wts repo, e.g. to expand documentation for JSON apis.
// cc @amihaiemil

Reply:
@ammaratef45 , @g4s8 , @amihaiemil , the following conversion is performed in `wts.rb` [here](https://github.com/zold-io/wts.zold.io/blob/7e3507fe96cd8cc6a0bd32a22ac1ab17644cc3e1/wts.rb#L464):
```
  amount = Zold::Amount.new(zld: params[:amount].to_f)
```
I am not a Ruby expert but my guess, which is based on [the documentation](https://apidock.com/ruby/String/to_f):
```
"123.45e1".to_f        #=> 1234.5
"45.67 degrees".to_f   #=> 45.67
"thx1138".to_f         #=> 0.0
```
is that
```
"amount": "1234.56"
```
would be interpretted as `1234.56` (Float).

Hope, that would help us out here. By the way, there has been opened a related to accurate decimal operations issue [#718](https://github.com/zold-io/zold/issues/718) in Zold project.